src/common.jl

```diff
@@ -1,80 +1,125 @@
-struct LinearCache{TA,Tb,Tp,Talg,Tc,Tr,Tl}
+struct LinearCache{TA,Tb,Tu,Tp,Talg,Tc,Tl,Tr}
     A::TA
     b::Tb
+    u::Tu
     p::Tp
     alg::Talg
-    cacheval::Tc
-    isfresh::Bool
-    Pr::Tr
-    Pl::Tl
+    cacheval::Tc  # store alg cache here 
+    isfresh::Bool # false => cacheval is set wrt A, true => update cacheval wrt A
+    Pl::Tl        # store final preconditioner here. not being used rn
+    Pr::Tr        # wrappers are using preconditioner in cache.alg for now
 end
 
-function set_A(cache, A)
+function set_A(cache::LinearCache, A)
     @set! cache.A = A
     @set! cache.isfresh = true
+    return cache
 end
 
-function set_b(cache, b)
+function set_b(cache::LinearCache, b)
     @set! cache.b = b
+    return cache
+end
+
+function set_u(cache::LinearCache, u)
+    @set! cache.u = u
+    return cache
 end
 
-function set_p(cache, p)
+function set_p(cache::LinearCache, p)
     @set! cache.p = p
-    # @set! cache.isfresh = true
+#   @set! cache.isfresh = true
+    return cache
 end
 
-function set_cacheval(cache::LinearCache, alg)
+function set_cacheval(cache::LinearCache, alg_cache)
     if cache.isfresh
-        @set! cache.cacheval = alg
+        @set! cache.cacheval = alg_cache
         @set! cache.isfresh = false
     end
     return cache
 end
 
-function SciMLBase.init(
-    prob::LinearProblem,
-    alg,
-    args...;
-    alias_A = false,
-    alias_b = false,
-    kwargs...,
-)
-    @unpack A, b, p = prob
-    if alg isa LUFactorization
-        fact = lu_instance(A)
-        Tfact = typeof(fact)
-    else
-        fact = nothing
-        Tfact = Any
-    end
-    Pr = nothing
-    Pl = nothing
+init_cacheval(alg::SciMLLinearSolveAlgorithm, A, b, u) = nothing
+
+function SciMLBase.init(prob::LinearProblem, alg, args...;
+                        alias_A = false, alias_b = false,
```

Comment:
these arguments should be handling the aliasing

Reply:
line 60-61 of the same file. 

```julia
    A = alias_A ? A : deepcopy(A)
    b = alias_b ? b : deepcopy(b)
```
This was in `common.jl` when I started the file.